src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -118,65 +262,704 @@ protected override void ProcessRecord()
+            }
+            // Ensure that the Passed in Uri has the same Host as the HttpClient. 
+            if (Uri.IsAbsoluteUri && httpClient.BaseAddress.Host != Uri.Host)
+            {
+                var error = GetValidationError($"Invalid Host {Uri.Host}", "InvokeGraphRequestInvalidHost",
+                    nameof(Uri));
+                ThrowTerminatingError(error);
+            }
+        }
+        /// <summary>
+        /// Validate Passed In Parameters
+        /// </summary>
+        private void ValidateParameters()
+        {
+            if (GraphRequestSession != null && SessionVariable != null)
+            {
+                var error = GetValidationError(
+                    "The cmdlet cannot run because the following conflicting parameters are specified: Session and SessionVariable. Specify either Session or SessionVariable, then retry.",
+                    "WebCmdletSessionConflictException");
+                ThrowTerminatingError(error);
+            }
+
             // When PATCH or POST is specified, ensure a body is present
-            if (this.Method == HttpVerb.PATCH || this.Method == HttpVerb.POST && this.Body == null)
+            if (Method == GraphRequestMethod.PATCH || Method == GraphRequestMethod.POST && Body == null)
+            {
+                var error = GetValidationError($"{nameof(Body)} is required when Method is {Method}",
+                    "InvokeGraphRequestBodyMissingWhenMethodIsSpecified", nameof(Body));
+                ThrowTerminatingError(error);
+            }
+
+            if (PassThru && OutFile == null)
+            {
+                var error = GetValidationError($"{nameof(OutFile)} is missing",
+                    "InvokeGraphRequestOutFileMissingException", nameof(PassThru));
+                ThrowTerminatingError(error);
+            }
+
+            if (Authentication == GraphRequestAuthenticationType.Default && !string.IsNullOrWhiteSpace(Token))
             {
-                ThrowTerminatingError(new ErrorRecord(new ArgumentNullException(nameof(Body), $"{nameof(this.Body)} is required when Method is {this.Method}"), Guid.NewGuid().ToString(),
-                    ErrorCategory.InvalidArgument, null));
+                var error = GetValidationError("AuthenticationTokenConflict",
+                    "WebCmdletAuthenticationTokenConflictException");
+                ThrowTerminatingError(error);
             }
+            // Token shouldn't be null when UserProvidedToken is specified
+            if (Authentication == GraphRequestAuthenticationType.UserProvidedToken && string.IsNullOrWhiteSpace(Token))
+            {
+                var error = GetValidationError("AuthenticationCredentialNotSupplied",
+                    "WebCmdletAuthenticationCredentialNotSuppliedException");
+                ThrowTerminatingError(error);
+            }
+
+            // Only Body or InFile can be specified at a time
+            if (Body != null && InFile != null)
```

Comment:
optional: You could validate the json if the Body is used by JToken.Parse(string)  and catch jsonreaderexception if it isn't valid. You have that IsJson implemented in the helpers.

Reply:
Since body can be json, byte array, a dictionary or a stream, just checking for null is enough.